services/auth/tests/url_builder_test.go

```diff
@@ -0,0 +1,109 @@
+package tests
+
+import (
+	"github.com/HackIllinois/api/services/auth/service"
+	"net/url"
+	"strings"
+	"testing"
+)
+
+const URL_SCHEME = "http"
+const URL_HOST = "hackillinois.com"
+const URL_PATH = "register/now"
+const CONSTRUCTED_BASIC_URL = "http://hackillinois.com/register/now"
+
+const DIFFERENT_SCHEME = "ftp"
+const HOST_WITH_PORT = "hackillinois.com:9800"
+const CONSTRUCTED_COMPLEX_URL = "ftp://hackillinois.com:9800/register/now"
+
+var QUERY_PARAMS = map[string]string{
+	"major":           "CS",
+	"graduating_year": "2021",
+}
+
+const EXPECTED_QUERY_STRING = "?graduating_year=2021&major=CS"
```

Comment:
Can you guarantee the ordering of the parameters in these expected query strings? As far as I know the functions which encode a `map[string]string` to a query string do not guarantee an ordering.

Reply:
I decided to fix this by testing the complete URL with just one query param, to guarantee the output. I also fixed the multi-param test by checking the url.ParseQuery() result against a known url.Values containing the params.